olp-cpp-sdk-authentication/src/AuthenticationClientImpl.cpp

```diff
@@ -357,15 +361,30 @@ TimeResponse AuthenticationClientImpl::ParseTimeResponse(
 }
 
 TimeResponse AuthenticationClientImpl::GetTimeFromServer(
-    client::CancellationContext context, const client::OlpClient& client) {
+    client::CancellationContext context,
+    const client::OlpClient& client) const {
   auto http_result = client.CallApi(kTimestampEndpoint, "GET", {}, {}, {},
                                     nullptr, {}, context);
 
   if (http_result.status != http::HttpStatusCode::OK) {
+    auto response = http_result.response.str();
+    OLP_SDK_LOG_WARNING_F(
+        kLogTag, "Failed to get time from server, status=%d, response='%s'",
+        http_result.status, response.c_str());
     return AuthenticationError(http_result.status, http_result.response.str());
   }
 
-  return ParseTimeResponse(http_result.response);
+  auto server_time = ParseTimeResponse(http_result.response);
+
+  if (!server_time) {
+    const auto& error = server_time.GetError();
```

Comment:
Why the error? Did you forget to add it to the log? I don't see it used otherwise.

Reply:
Used on the next line

Reply:
Ahh, yes. Forgot that we have different error structs. My bad, sorry.